spec/features/sign_in_spec.rb

```diff
@@ -0,0 +1,55 @@
+require 'rails_helper'
+
+feature "Vistor signs in to their account" do
+  scenario "User enters valid information to sign in" do
+    #will be replaced with factory girl
+    visit root_path
+    click_link 'Sign Up'
+
+    fill_in 'First Name', with: 'Jon'
+    fill_in 'Last Name', with: 'Smith'
+    fill_in 'Email', with: 'user@email.com'
+    fill_in 'user_password', with: 'password'
+    fill_in 'Password Confirmation', with: 'password'
+    click_button 'Sign Up'
+    click_link 'Sign Out'
+    # visit root_path
+    click_link 'Sign In'
+
+    fill_in 'Email', with: 'user@email.com'
+    fill_in 'user_password', with: 'password'
+    click_button 'Sign In'
+
+    expect(page).to have_content("Sign Out")
+    expect(page).to_not have_content("Sign In")
+    expect(page).to have_content("Signed in successfully.")
+  end
+
+  scenario "User enters invalid information" do
+    visit root_path
+    click_link 'Sign In'
+
+    fill_in 'Email', with: 'notauser@email.com'
+    fill_in 'user_password', with: 'wrongpassword'
+    click_button 'Sign In'
+
+    expect(page).to have_content("Sign In")
+    expect(page).to_not have_content("Sign Out")
+    expect(page).to have_content("Invalid Email or password.")
+  end
+
+  # uncomment when factory girl is set up
+  # scenario "User signs out" do
+  #   visit root_path
+  #   click_link 'Sign In'
+  #
+  #   fill_in 'Email', with: 'user@email.com'
+  #   fill_in 'user_password', with: 'password'
+  #   click_button 'Sign In'
+  #   click_link 'Sign Out'
+  #
+  #   expect(page).to have_content('Sign In')
+  #   expect(page).to_not have_content("Sign Out")
+  #   expect(page).to have_content("Signed out successfully.")
+  # end
```

Comment:
Please try not to leave commented out code in a PR

Reply:
It's commented out because its a good test but doesnt really work without factory girl.  I'll make it work and repush